pkg/cluster/deploybaseresources_test.go

```diff
@@ -400,3 +402,209 @@ func TestEnsureInfraID(t *testing.T) {
+						ServiceEndpoints: &[]mgmtnetwork.ServiceEndpointPropertiesFormat{
+							{
+								Service: &serviceEndpoint,
+								Locations: &[]string{
+									"*",
+								},
+							},
+						},
+					},
+				}
+
+				subnet.EXPECT().Get(ctx, masterSubnet).Return(subnetWithServiceEndpoint, nil)
+				subnet.EXPECT().Get(ctx, fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001")).Return(subnetWithServiceEndpoint, nil)
+			},
+			workerSubnets: []string{
+				fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001"),
+			},
+			wantSubnets: []string{
+				masterSubnet,
+				fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001"),
+			},
+		},
+		{
+			name: "mixed subnets with service endpoint",
+			mocks: func(subnet *mock_subnet.MockManager) {
+				subnetWithServiceEndpoint := &mgmtnetwork.Subnet{
+					SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+						ServiceEndpoints: &[]mgmtnetwork.ServiceEndpointPropertiesFormat{
+							{
+								Service: &serviceEndpoint,
+								Locations: &[]string{
+									location,
+								},
+							},
+						},
+					},
+				}
+
+				subnet.EXPECT().Get(ctx, masterSubnet).Return(subnetWithServiceEndpoint, nil)
+				subnet.EXPECT().Get(ctx, fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001")).Return(subnetWithServiceEndpoint, nil)
+				subnet.EXPECT().Get(ctx, fmt.Sprintf(workerSubnetFormatString, "worker-subnet-002")).Return(&mgmtnetwork.Subnet{}, nil)
+			},
+			workerSubnets: []string{
+				fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001"),
+				fmt.Sprintf(workerSubnetFormatString, "worker-subnet-002"),
+				"",
+			},
+			wantSubnets: []string{
+				masterSubnet,
+				fmt.Sprintf(workerSubnetFormatString, "worker-subnet-001"),
+			},
+		},
+		{
+			name: "Get subnet returns error",
+			mocks: func(subnet *mock_subnet.MockManager) {
+				subnet.EXPECT().Get(ctx, masterSubnet).Return(nil, errors.New("generic error"))
+			},
+			workerSubnets: []string{},
+			wantErr:       "generic error",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			subnet := mock_subnet.NewMockManager(controller)
+			tt.mocks(subnet)
+
+			workerProfiles := []api.WorkerProfile{}
+			if tt.workerSubnets != nil {
+				for _, subnet := range tt.workerSubnets {
+					workerProfiles = append(workerProfiles, api.WorkerProfile{
+						SubnetID: subnet,
+					})
+				}
+			}
+
+			m := &manager{
+				doc: &api.OpenShiftClusterDocument{
+					Key: strings.ToLower(resourceID),
+
+					OpenShiftCluster: &api.OpenShiftCluster{
+						ID:       resourceID,
+						Name:     "FoobarCluster",
+						Location: location,
+
+						Properties: api.OpenShiftClusterProperties{
+							MasterProfile: api.MasterProfile{
+								SubnetID: masterSubnet,
+							},
+							WorkerProfiles: workerProfiles,
+						},
+					},
+				},
+				subnet: subnet,
+			}
+
+			// hopefully setting a seed here means it passes consistently :)
```

Comment:
nothing like betting on rng

Reply:
:game_die: 

I don't think we actually need seeding since we're not using random at all.  I'll remove.